retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
We ( @fabxc @juliusv @beorn7 ) discussed this a while in person.

Currently, we tend to go for @brian-brazil 's idea of attaching an empty `job` and `instance` label to every metric without such label(s) on the `/federation` endpoint. Then, we need a neat config options for the scrape config. See https://github.com/prometheus/prometheus/issues/490 for further discussion. I'll update said issue in a sec.